### DISCUSSION TITLE
Allow long error messages on a single line

### DISCUSSION THREAD

**Discussion by llamafilm** [curl/curl#13804] [Ideas]
Sometimes a long error message is split into 2 lines.  This makes it harder to read IMHO, and also hard to parse with a script.  Example:

```
% curl -fD - --retry 5 --retry-all-errors https://google.com/asdfasdf 
HTTP/2 404 
content-type: text/html; charset=UTF-8
referrer-policy: no-referrer
content-length: 1569
date: Tue, 28 May 2024 00:33:21 GMT
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000

curl: (22) The requested URL returned error: 404
Warning: Problem (retrying all errors). Will retry in 1 seconds. 5 retries 
Warning: left.
```

 It looks like it assumes a default terminal width of 79 characters. https://github.com/curl/curl/blob/53b4dfe405dca1eeff0ac48e2fc96d7782e887a7/src/tool_msgs.c#L49

Can this behavior be adjusted?  Perhaps use the existing `COLUMNS` variable or a new one.

**Comment by bagder** [comment]
Ack, it should check the terminal width. I'll give it a look...

**Reply by bagder** [reply]
See https://github.com/curl/curl/pull/13808

**Reply by llamafilm** [reply]
Thank you!

**Comment by bagder** [comment]
Merged now, to be included in the next release.
